### DISCUSSION TITLE
How to get SOC?

### DISCUSSION THREAD

**Discussion by daath** [johanmeijer/grott#330] [General]
I run Grott with two inverters and run grott_ha - in the logs of Grott I can see that it sends the SOC:

```
	 - MQTT jsonmsg: 
                        {{ ..., "vbat": 541, "SOC": 99, "pactouserr":...}}
```

and in grott_ha.py I see:

```
    "SOC": {
        "name": "Battery charge",
        "device_class": "battery",
        "state_class": "measurement",
        "unit_of_measurement": "%",
        "icon": "mdi:battery-charging-60",
    },

```

But I can't seem to find it in the sensors? How do I access this?

**Comment by daath** [comment]
I just noticed that it has a duplicate name?

```
    "p1charge1": {
        "name": "Battery charge",
        "device_class": "power",
        "unit_of_measurement": "kW",
        "state_class": "measurement",
        "icon": "mdi:battery-arrow-up",
    },

```

Both is called "Battery charge" and it's the last one that gets shown in the sensors?

**Comment by daath** [comment]
I am so confused.

I changed:

```
"SOC": {
        "name": "Battery charge",
        "device_class": "battery",
        "state_class": "measurement",
        "unit_of_measurement": "%",
        "icon": "mdi:battery-charging-60",
    },
```

To: 

```
    "SOC": {
        "name": "Battery SOC",
        "device_class": "battery",
        "state_class": "measurement",
        "unit_of_measurement": "%",
        "icon": "mdi:battery-charging-60",
    },
```

Restarted Grott and now SOC showed up in the sensors, but with the name `sensor.deviceserial_battery_charge_2`?! It was definitely NOT there bedore ... (Obviously deviceserial is the serial of my inverter)... 

Oh, and its value is 0.99 and 1 etc - so I of course needed to * 100 :)

**Reply by egguy** [reply]
It seems it's the old plugin. I need to create a more recent pull request.

There was also a pull request a few weeks ago to correct the divider.

If you use the add-on I recommend you to use the edge version which contains the latest updates.

**Reply by daath** [reply]
I use the docker version and I just downloaded grott_ha.py - did I download an old one?! Where do I get the new one?

**Reply by shampeon** [reply]
The `_2` might be an artifact of how HA handles entity names. If it encounters duplicate entity names it appends numbers to them.

But in this case `p1charge1` is a measurement in kW, and `SOC` is a percentage so they're different measurements.

**Reply by shampeon** [reply]
@egguy Where is the latest add-on located? I have a different [question open](https://github.com/johanmeijer/grott/discussions/326) about using grott_ha. I'm using 0.0.7-rc6, the one checked into this repo.

**Comment by daath** [comment]
No idea what is happening - now SOC disappeared from the sensors card, but I can still access it with the DEVICEID_battery_charge_2 name ... What is going on? 🤣
